### PR TITLE
bug/renamed-columns

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- internal ticket

**This PR will result in the following new package version:**

- v1.1.0 - breaking since some datatypes can change and could affect customers' downstream uses.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## 🚨 Breaking Change 🚨
> - This change is made breaking due to changes made in the source package. See the [v1.1.0 dbt_salesforce_source release notes](https://github.com/fivetran/dbt_salesforce_source/releases/tag/v1.0.1) for more details.
> - Added logic to support user-specified scenarios where the Fivetran Salesforce connector syncs column names using the original Salesforce API naming convention. For example, while Fivetran typically provides the column as `created_date`, some users might choose to receive it as `CreatedDate` according to the API naming. This update ensures the package is automatically compatible with both naming conventions.
>   - Specifically, the package now performs a COALESCE, preferring the original Salesforce API naming. If the original naming is not present, the Fivetran version is used instead.
>   - Renamed columns are now explicitly cast to prevent conflicts during the COALESCE. 
>   - ❗This change is considered breaking since the resulting column types may differ from prior versions of this package.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
    
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- I had tried to write consistency tests, but because of the datatype changes necessitating the breaking change, comparing rows did not make sense. I extended the column test from the source package to look at all tables and column names, and that passed.
- The other way I tested was running the transform package and made sure it ran without error and brought data through.

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: packages.yml
Comment (fivetran-catfritz) [inline packages.yml:6]:
```suggestion
- package: fivetran/salesforce_source
  version: [">=1.1.0", "<1.2.0"]
```
commit before merge

File: integration_tests/tests/consistency/consistency_columns.sql
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/consistency/consistency_columns.sql:1]: Thanks for creating this here as well!

File: integration_tests/tests/consistency/consistency_columns.sql
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/consistency/consistency_columns.sql:18]: Now that we are making this a breaking change, would it make more sense to remove this case when statement? We know it will fail for this release (why it is breaking now), but it should not need this case statement for future updates and could risk us not catching datatype mismatches.

File: integration_tests/dbt_project.yml
Comment (fivetran-joemarkiewicz) [inline integration_tests/dbt_project.yml:2]: I know this wasn't part of your update, but in order for your `consistency_columns` validation to properly work across both the source and the transform package, we will want to change the scope of the `+schema:` models config. Otherwise, when I run this locally I see numerous failures because my staging models are not being materialized in the `*_dev` schema.

Therefore, we will need to change lines 7-11 to be the following:
```yml
models:
  +materialized: table
  +schema: "salesforce_{{ var('directed_schema','dev') }}"
```

Comment (fivetran-joemarkiewicz) [reply integration_tests/dbt_project.yml:2]: For example, without this change I see 304 row failures which is not expected 😱 

image

File: integration_tests/tests/consistency/consistency_columns.sql
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/consistency/consistency_columns.sql:44]: This test is great! However, if there is a failure it is really difficult to decipher. Which record is from the prod/dev test and which is from the dev/prod test. Would there be a way to make this easier to understand when there is a failure?

Not required in this update, but I know we are reusing this a lot and would like to think about this for a future enhancement.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz thanks for applying this breaking change update here and also adding the validation test! A have a few small comments and one small thing I wanted to callout, I noticed the `consistency_daily_activity` test for me is failing when using our internal data 🤔

image

Would you be able to look into this and see if you also experience these row failures? At first glance it looks like the only field are not tying out between dev and prod is the following:
- tasks_completed

File: integration_tests/tests/consistency/consistency_columns.sql
Comment (fivetran-catfritz) [reply integration_tests/tests/consistency/consistency_columns.sql]: Good point. I found a way and updated!

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-joemarkiewicz I have made the update here as well!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM once the package dep is bumped

File: CHANGELOG.md
Comment (fivetran-avinash) [inline CHANGELOG.md:5]:
```suggestion
- This change is made breaking due to changes made in the source package. See the [v1.1.0 dbt_salesforce_source release notes](https://github.com/fivetran/dbt_salesforce_source/releases/tag/v1.1.0) for more details.
```

File: CHANGELOG.md
Comment (fivetran-avinash) [inline CHANGELOG.md:10]: Would recommend adding a Under the Hood section, and a line for the consistency tests you created.

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-catfritz Ready for release on Monday, with two minor edits on the CHANGELOG!

File: CHANGELOG.md
Comment (fivetran-catfritz) [reply CHANGELOG.md:5]: Ooo good good catch.

**Pull request merged**
